app/Main.hs

```diff
@@ -84,22 +86,27 @@ vaultRetryPolicy opts = Retry.fullJitterBackoff (unMilliSeconds (oRetryBaseDelay
 main :: IO ()
 main = do
   localEnvVars <- getEnvironment
-  cliAndEnvOptions <- parseOptionsFromEnvAndCli localEnvVars
+  envFileSettings <- readConfigFromEnvFiles
 
-  if (oLogLevel cliAndEnvOptions) <= Info
-    then print cliAndEnvOptions
+  -- Deduplicate, give precedence to set env vars over .env files
+  let envAndEnvFileConfig = nubBy (\(x, _) (y, _) -> x == y) localEnvVars ++ envFileSettings
```

Comment:
Do you want to deduplicate the `localEnvVars` and then concatenate, or did you mean to deduplicate after concatenating? In either case, adding explicit parentheses here could help clear things up.

Also, FYI (but the way it's written now is fine and clear), there's `Data.Function.on` which allows you to write the custom equality you wrote as ``(==) `on` fst``.

Reply:
Done